chain/types/actor_event.go

```diff
@@ -0,0 +1,47 @@
+package types
+
+import (
+	"github.com/ipfs/go-cid"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+)
+
+type ActorEventBlock struct {
+	// what value codec does client want to match on ?
+	Codec uint64 `json:"codec"`
```

Comment:
How about this - we only want `Codec` to be one of two types at the moment, and I think we ought to be in control of that. We also have a type for codecs, so we could do this:

```suggestion
	"github.com/ipfs/go-cid"
	"github.com/multiformats/go-multicodec"

	"github.com/filecoin-project/go-address"
	"github.com/filecoin-project/go-state-types/abi"
)

type EventCodec multicodec.Code

// Currently supported codecs for actor event data
const (
	Raw  EventCodec = EventCodec(multicodec.Raw)
	Cbor EventCodec = EventCodec(multicodec.Cbor)
)

type ActorEventBlock struct {
	// what value codec does client want to match on ?
	Codec EventCodec `json:"codec"`
```

Reply:
yeah, well since these are for the JSON API maybe it's not that much of a change since these will be mostly constructed by the JSON decoder :shrug: probably not a big improvement